includes/Admin/Google_Product_Category_Field.php

```diff
@@ -19,147 +19,21 @@
  */
 class Google_Product_Category_Field {
 
-
-	/** @var string the WordPress option name where the full categories list is stored */
-	const OPTION_GOOGLE_PRODUCT_CATEGORIES = 'wc_facebook_google_product_categories';
-
-
 	/**
 	 * Instantiates the JS handler for the Google product category field.
 	 *
 	 * @since 2.1.0
 	 *
-	 * @param string $input_id element that should receive the latest concrete category ID value
+	 * @param string $input_id element that should receive the latest concrete category ID value.
 	 */
 	public function render( $input_id ) {
-
-		$js = sprintf(
+		$facebook_category_handler = facebook_for_woocommerce()->get_facebook_category_handler();
+		$facebook_category_fields  = sprintf(
 			"window.wc_facebook_google_product_category_fields = new WC_Facebook_Google_Product_Category_Fields( %s, '%s' );",
-			json_encode( $this->get_categories() ),
+			json_encode( $facebook_category_handler->get_categories() ),
 			esc_js( $input_id )
 		);
 
-		wc_enqueue_js( $js );
-
-	}
-
-
-	/**
-	 * Gets the full categories list from Google and stores it.
-	 *
-	 * @since 2.1.0
-	 */
-	public function get_categories() {
-
-		// only fetch again if not fetched less than one hour ago
-		$categories = get_transient( self::OPTION_GOOGLE_PRODUCT_CATEGORIES );
-
-		if ( empty ( $categories ) ) {
-
-			// fetch from the URL
-			$categories_response = wp_remote_get( 'https://www.google.com/basepages/producttype/taxonomy-with-ids.en-US.txt' );
-
-			$categories = $this->parse_categories_response( $categories_response );
-
-			if ( ! empty( $categories ) ) {
-
-				set_transient( self::OPTION_GOOGLE_PRODUCT_CATEGORIES, $categories, WEEK_IN_SECONDS );
-				update_option( self::OPTION_GOOGLE_PRODUCT_CATEGORIES, $categories, 'no' );
```

Comment:
@budzanowski we should delete the transient and the option from the database in the 2.3.4 update routine.

Reply:
Good idea.

Reply:
Done in: e9a66a0b68d53cc37085ca65efb864759df024ca